### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by natkhosh**
Tasks 1-5 --> done

### CODE REVIEW

File: homework1/task_2/task_2.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Given a cell with "it's a fib sequence" from slideshow,
+please write function "check_fib", which accepts a Sequence of integers, and
+returns if the given sequence is a Fibonacci sequence
+
+We guarantee, that the given sequence contain >= 0 integers inside.
+"""
+from typing import Sequence
+
+
+def _check_window(x: int, y: int, z: int) -> bool:
```
Comment (lev-titov) [inline homework1/task_2/task_2.py]: what is a window?
and why function starts with a _?

File: homework1/task_2/task_2.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    """
+    Support function to checking sum
+    :param x: integer
+    :param y: integer
+    :param z: integer
+    :return: bool
+    """
+    return (x + y) == z
```
Comment (lev-titov) [inline homework1/task_2/task_2.py]: *optional*
rudimentary parentheses

File: homework1/task_2/task_2.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def check_fibonacci(data: Sequence[int]) -> bool:
+    """
+    Function for check a Fibonacci sequence.
+    :param data: Sequence of integers
+    :return: bool: true for success if the given sequence is a Fibonacci sequence, False otherwise.
+    """
+    if len(data) < 3:
```
Comment (lev-titov) [inline homework1/task_2/task_2.py]: why sequence with 2 elements is invalid?
[2, 3] is valid subsequence

File: homework1/task_2/task_2.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        print("Invalid data")
+        return False
+    else:
+        while len(data) >= 3:
+            x, y, z = data[0], data[1], data[2]
+
+            if _check_window(x, y, z):
```
Comment (lev-titov) [inline homework1/task_2/task_2.py]: this function only checks if z = x + y
but its not a only sign of Fibonacci sequence
for example in your case [10, -5, 5] will return true, but its not a Fibonacci

File: homework1/task_2/task_2.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+def check_fibonacci(data: Sequence[int]) -> bool:
+    """
+    Function for check a Fibonacci sequence.
+    :param data: Sequence of integers
+    :return: bool: true for success if the given sequence is a Fibonacci sequence, False otherwise.
+    """
+    if len(data) < 3:
+        print("Invalid data")
+        return False
+    else:
```
Comment (lev-titov) [inline homework1/task_2/task_2.py]: remove else

File: homework1/task_5/task_5.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+        for i in range(0, arr_len - k + 1):
+            for j in range(2, k + 1):
+                # Sum definition, the aggregate of two or more numbers
+                subarray_sum = sum(nums[i : i + j])
+                if subarray_sum > max_sum:
+                    max_sum = subarray_sum
+        return max_sum
+    else:
+        print("Argument Error: nums length must be >= 2, k must be >= 2")
```
Comment (lev-titov) [inline homework1/task_5/task_5.py]: why do you add extra conditions?
both nums and k can be 1

**Review by lev-titov — CHANGES_REQUESTED** [review]

File: homework1/task_5/task_5.py (hunk 6 above)
Comment (natkhosh) [reply homework1/task_5/task_5.py]: >> both nums and k can be 1
'Sum' definition: the aggregate of two or more numbers. In this case i suppose to exclude nums and k < 2

**Comment by natkhosh** [comment]
task5: fixed
task2: changed the algorithm

File: homework1/task_2/task_2.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+def check_fibonacci(data: Sequence[int]) -> bool:
+    """
+    Function for check a Fibonacci sequence.
+    :param data: Sequence of integers
+    :return: bool: true for success if the given sequence is a Fibonacci sequence, False otherwise.
+    """
+    if len(data) < 2:
+        return False
+    elif len(data) < 3:
```
Comment (lev-titov) [inline homework1/task_2/task_2.py]: elif -> if

File: homework1/task_2/task_2.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        for i in range(1, len(data)):
+            if not (
+                (data[i - 1] <= data[i])
+                and isFibonacci(data[i - 1])
+                and isFibonacci(data[i])
+                and isFibonacci(data[i - 1] + data[i])
+            ):
+                return False
+        return True
+    else:
```
Comment (lev-titov) [inline homework1/task_2/task_2.py]: remove else

File: homework1/task_5/task_5.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    """
+    Function find a sub-array with length less then or equal to "k", with maximal sum.
+    :param nums: list of integer numbers
+    :param k: length of sub-array
+    :return: maximal sum of sub-array
+    """
+    arr_len = len(nums)
+    # max_sum = 0
```
Comment (lev-titov) [inline homework1/task_5/task_5.py]: ?

File: homework1/task_5/task_5.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+    if arr_len < 1:
+        return 0
+
+    max_sum = max(nums)
+    if k == 1:
```
Comment (lev-titov) [inline homework1/task_5/task_5.py]: this condition is useless, just remove it and nothing will change

File: homework1/task_2/task_2.py [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+def check_fibonacci(data: Sequence[int]) -> bool:
+    """
+    Function for check a Fibonacci sequence.
+    :param data: Sequence of integers
+    :return: bool: true for success if the given sequence is a Fibonacci sequence, False otherwise.
+    """
+    if len(data) < 2:
+        return False
+    elif len(data) < 3:
+        for i in range(1, len(data)):
```
Comment (lev-titov) [inline homework1/task_2/task_2.py:42]: do u really need this cycle considering that we go here only if len = 2?

File: homework1/task_5/task_5.py [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    Function find a sub-array with length less then or equal to "k", with maximal sum.
+    :param nums: list of integer numbers
+    :param k: length of sub-array
+    :return: maximal sum of sub-array
+    """
+    arr_len = len(nums)
+    # max_sum = 0
+
+    if arr_len < 1:
```
Comment (lev-titov) [inline homework1/task_5/task_5.py]: *optional*
```suggestion
    if not arr_len :
```

**Review by lev-titov — CHANGES_REQUESTED** [review]

**Review by lev-titov — APPROVED** [review]

**Pull request merged**